subsys/net/ip/net_context.c

```diff
@@ -399,65 +399,74 @@ int net_context_get(sa_family_t family,
 
 #if defined(CONFIG_NET_CONTEXT_CHECK)
 
+	switch(family) {
+	case AF_INET:
 #if !defined(CONFIG_NET_IPV4)
-	if (family == AF_INET) {
 		NET_ASSERT_INFO(family != AF_INET, "IPv4 disabled");
 		return -EPFNOSUPPORT;
-	}
 #endif
+		break;
 
+	case AF_INET6:
 #if !defined(CONFIG_NET_IPV6)
-	if (family == AF_INET6) {
 		NET_ASSERT_INFO(family != AF_INET6, "IPv6 disabled");
 		return -EPFNOSUPPORT;
-	}
 #endif
+		break;
+
+	default:
+		NET_ASSERT_INFO(family == AF_INET || family == AF_INET6,
+				"Unknown address family %d", family);
+		return -EAFNOSUPPORT;
+	}
 
+	switch(type) {
+	case SOCK_DGRAM:
 #if !defined(CONFIG_NET_UDP)
-	if (type == SOCK_DGRAM) {
 		NET_ASSERT_INFO(type != SOCK_DGRAM,
```

Comment:
ok so legacy code is using an ASSERT where NET_ERR/NET_WARN could have been sufficient.